madmom/ml/nn/layers.py

```diff
@@ -23,9 +23,16 @@ class Layer(object):
 
     """
 
-    def __call__(self, *args):
+    def __call__(self, *args, **kwargs):
         # this magic method makes a Layer callable
-        return self.activate(*args)
+        return self.activate(*args, **kwargs)
+
+    def __getstate__(self):
```

Comment:
IMHO, this shouldn't be in the `Layer` class. Every subclass of `Layer` should be responsible to ensure that their state is not pickled. This does not lead to much code duplication - right now, only `RecurrentLayer` and `LSTMLayer` are stateful, and I think for `LSTMLayer` you'll have to overwrite `__getstate__` anyways, because you don't want to pickle the previous state and the cell state.